packages/evm/contracts/Settler.sol

```diff
@@ -312,6 +313,20 @@ contract Settler is ISettler, Ownable, ReentrancyGuard, EIP712 {
             if (proposalFee > maxFee) revert SettlerSolverFeeTooHigh(proposalFee, maxFee);
         }
 
+        uint8 minValidations = IController(controller).minValidations();
+        uint256 requiredValidations = intent.minValidations > minValidations ? intent.minValidations : minValidations;
+
+        if (intent.validations.length < requiredValidations) {
+            revert SettlerIntentValidationsNotEnough(requiredValidations, intent.validations.length);
+        }
+
+        for (uint256 i = 0; i < intent.validations.length; i++) {
+            Validation memory validation = Validation(intent.hash());
+            address validator = ECDSA.recover(_hashTypedDataV4(validation.hash()), intent.validations[i]);
+            bool isValidatorNotAllowed = !IController(controller).isValidatorAllowed(validator);
+            if (isValidatorNotAllowed) revert SettlerValidatorNotAllowed(validator);
+        }
```

Comment:
We should also check all the requested addresses are different, meaning, from different validators